cmd/logs.go

```diff
@@ -26,106 +28,149 @@ import (
+		Aliases:   []string{"log"},
+	}
+	cmd.Flags().Int64Var(&flags.tail, "tail", flags.tail, "Lines of recent log file to display. Defaults to 100 with no selector, showing all log lines otherwise 10, if a selector is provided.")
+	cmd.Flags().DurationVar(&flags.sinceSeconds, "since", flags.sinceSeconds, "Only return logs newer than a relative duration like 5s, 2m, or 3h. Defaults to all logs. Only one of since-time / since may be used.")
+	cmd.Flags().StringVar(&flags.sinceTime, "since-time", flags.sinceTime, "Only return logs after a specific date (RFC3339). Defaults to all logs. Only one of since-time / since may be used.")
+	cmd.Flags().BoolVar(&flags.elasticsearch, "elasticsearch", flags.elasticsearch, "Show logs from elasticsearch")
+	return cmd
+}
+
+func validateArgs(args []string) {
+	if len(args) < 1 || len(args) > 2 {
+		fmt.Println("Command must be in the format: logs (gardener-apiserver|gardener-controller-manager|ui|api|scheduler|controller-manager|etcd-operator|etcd-main[etcd backup-restore]|etcd-events[etcd backup-restore]|addon-manager|vpn-seed|vpn-shoot|machine-controller-manager|dashboard|prometheus|grafana|alertmanager|tf (infra|dns|ingress) args(--elasticsearch|--tail|--since|--since-time|--timestamps)")
+		os.Exit(2)
+	}
+	var t Target
+	ReadTarget(pathTarget, &t)
+	if len(t.Target) < 3 && (args[0] != "gardener-apiserver") && (args[0] != "gardener-controller-manager") && (args[0] != "tf") && (args[0] != "dashboard") {
+		fmt.Println("No shoot targeted")
+		os.Exit(2)
+	} else if (len(t.Target) < 2 && (args[0] == "tf")) || len(t.Target) < 3 && (args[0] == "tf") && (t.Target[1].Kind != "seed") {
+		fmt.Println("No seed or shoot targeted")
+		os.Exit(2)
+	} else if len(t.Target) == 0 {
+		fmt.Println("Target stack is empty")
+		os.Exit(2)
+	}
+}
+
+func validateFlags(flags *logFlags) {
+
+	if flags.sinceSeconds != 0 && flags.sinceTime != "" {
+		fmt.Println("Logs command can not contains --since and --since-time in the same time")
+		os.Exit(2)
+	} else if flags.sinceTime != "" {
+		value, err := time.Parse(time.RFC3339, flags.sinceTime)
+		if err != nil {
+			fmt.Println("Incorrect value for flag: --since-time")
 			os.Exit(2)
+		} else {
+			flags.sinceSeconds = time.Since(value)
 		}
-		var t Target
-		ReadTarget(pathTarget, &t)
-		if len(t.Target) < 3 && (args[0] != "gardener-apiserver") && (args[0] != "gardener-controller-manager") && (args[0] != "tf") && (args[0] != "dashboard") {
-			fmt.Println("No shoot targeted")
-			os.Exit(2)
-		} else if (len(t.Target) < 2 && (args[0] == "tf")) || len(t.Target) < 3 && (args[0] == "tf") && (t.Target[1].Kind != "seed") {
-			fmt.Println("No seed or shoot targeted")
-			os.Exit(2)
-		} else if len(t.Target) == 0 {
-			fmt.Println("Target stack is empty")
-			os.Exit(2)
+	}
+}
+
+func runCommand(args []string) {
+	switch args[0] {
+	case "gardener-apiserver":
```

Comment:
We could also add `--elasticsearch` to support gardener-apiserver, gardener-controller-manager.

Reply:
Ignore this one, sorry, my mistake.